task-service/src/main/java/com/infodation/task_service/components/JwtAuthenticationFilter.java

```diff
@@ -0,0 +1,116 @@
+package com.infodation.task_service.components;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.http.*;
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.authority.SimpleGrantedAuthority;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.stereotype.Component;
+import org.springframework.util.StringUtils;
+import org.springframework.web.client.HttpClientErrorException;
+import org.springframework.web.client.RestTemplate;
+import org.springframework.web.filter.OncePerRequestFilter;
+
+import javax.servlet.FilterChain;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Component
+@Slf4j
+public class JwtAuthenticationFilter extends OncePerRequestFilter {
+    private final RestTemplate restTemplate;
+    private final ObjectMapper mapper;
+    @Value("${auth.service.baseUrl}")
+    private String authServiceUrl;
+
+    public JwtAuthenticationFilter(RestTemplate restTemplate, ObjectMapper mapper) {
+        this.restTemplate = restTemplate;
+        this.mapper = mapper;
+    }
+
+    @Override
+    protected void doFilterInternal(HttpServletRequest request,
+                                    HttpServletResponse response, FilterChain filterChain)
+            throws ServletException, IOException {
+
+        ResponseEntity<String> validationResponse = null;
+        response.setContentType("application/json");
+        response.setCharacterEncoding("UTF-8");
+        String message = null;
+        int status = HttpServletResponse.SC_OK;
+        try {
+            String jwt = request.getHeader(HttpHeaders.AUTHORIZATION);
+            if (jwt == null || !jwt.startsWith("Bearer ")) {
```

Comment:
Move the `"Bearer "` value into an constant.
```suggestion
            private final String BEAR_TOKEN_PREFIX = "Bearer "
```